lib/streamlit/DeltaGenerator.py

```diff
@@ -1365,11 +1373,27 @@ def video(self, element, data, format="video/mp4", start_time=0):
         # TODO: Provide API to convert raw NumPy arrays to video file (with
         # proper headers, etc)?
         import streamlit.elements.generic_binary_proto as generic_binary_proto
+        from validators import url
 
-        generic_binary_proto.marshall(element.video, data)
         element.video.format = format
         element.video.start_time = start_time
 
+        if isinstance(data, string_types) and url(data):
+            # If this is a straight YouTube link, reformat it into an embed link
+            # before it gets shipped to the frontend.
+            # 
+            # Regex covers any youtube URL (incl. shortlinks) that points to a
+            # valid video but needs to be converted to an "embed" link.
+            re_yt_watchlink = re.compile(
+                "http(?:s?):\/\/(?:www\.)?youtu(?:be\.com\/watch\?v=|\.be\/)(?P<code>[\w\-\_]*)(&(amp;)?[\w\?=]*)?")
```

Comment:
Did you take this regex from somewhere? Stackoverflow, etc?
If so, please link it in a comment for future reference.

Reply:
Hmm... I can make a regexr.com entry and link to that?

When I refactor the Youtube parts to a utility function, it will get its own test with a nice set of fixtures to demonstrate its coverage.

Reply:
https://regexr.com/4n2h1